runtime/onert/core/src/util/shapeinf/Onehot.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape onehotShape(const ir::Shape &input_shape, const int depth, const int axis)
+{
+  int rank = input_shape.rank();
+
+  ir::Shape newShape(rank + 1); // output shape rank will be input rank + 1.
+
+  // ToDo : It will consider that case of the indice was ragged tensor.
+  if (rank == 1)
+  { // It means inputed indice is scalar.
+    int features = input_shape.num_elements();
+
+    if (axis == 0)
+    { // new_shape will be [batch] x [features].
+      newShape.dim(0) = depth;
+      newShape.dim(1) = features;
+    }
+    else
+    { // new_shape will be [features] x [depth].
+      newShape.dim(0) = features;
+      newShape.dim(1) = depth;
+    }
+  }
+  else if (rank == 2)
+  { // it means inputed indice is matrix(batch).
+    int batch = input_shape.dim(0);
+    int features = input_shape.dim(1);
+
+    if (axis == -1)
+    { // new_shape will be [batch] x [features] x [depth].
+      newShape.dim(0) = batch;
+      newShape.dim(1) = features;
+      newShape.dim(2) = depth;
+    }
+    else if (axis == 1)
+    { // new_shape will be [batch] x [depth] x [features].
+      newShape.dim(0) = batch;
+      newShape.dim(1) = depth;
+      newShape.dim(2) = features;
+    }
+    else
+    { // axis == 0
+      // new_shape will be [depth] x [batch] x [features].
+      newShape.dim(0) = depth;
+      newShape.dim(1) = batch;
+      newShape.dim(2) = features;
+    }
+  }
```

Comment:
How about changing it like below?

```suggestion
  axis += newShape.rank();
  for (size_t i = 0; i < newShape.rank(); ++i) {
    if (i < axis) {
      newShape.dim(i) = input_shape.dim(i);
    } else if (i == axis) {
      newShape.dim(i) = depth;
    } else {
      newShape.dim(i) = input_shape.dim(i - 1); 
    }   
  }
```

Reply:
It has been updated.